actioncable/lib/action_cable/connection/subscriptions.rb

```diff
@@ -46,7 +46,7 @@ def remove_subscription(subscription)
       end
 
       def perform_action(data)
-        find(data).perform_action(decode_hash(data['data']))
+        find(data).perform_action ActiveSupport::JSON.decode(data['data'])
```

Comment:
Gotcha. There's no new payload; it's the same format. Permitting non-JSON-encoded `identifier` and `data` values was enabled for a while on master, but it hasn't been released and none of our tooling uses it. I'll reference the original PRs and issues to close the loop.